backup/backup.go

```diff
@@ -72,3 +75,102 @@ func Backup(rawPaths []string, serviceCh UploadServiceFactory) {
 	wg.Wait()
 	log.Println("Backup complete")
 }
+
+type file_status struct {
+	file File
+	// no enums lol
+	modified bool
+	new      bool
+}
+
+func compareFileToDb(path string, info os.FileInfo, tx *sql.Tx) file_status {
+	var expectedLastModifiedTime int64
+	var expectedSize int64
+	size := info.Size()
+	err := tx.QueryRow("SELECT files.fs_modified, sizes.size FROM files INNER JOIN sizes ON files.hash = sizes.hash WHERE files.path = ? AND files.end IS NULL", path).Scan(&expectedLastModifiedTime, &expectedSize)
+	if err == nil {
+		if expectedLastModifiedTime != info.ModTime().Unix() || expectedSize != size {
+			return file_status{file: File{path, info}, modified: true}
+		} else {
+			return file_status{file: File{path, info}}
+		}
+	} else {
+		if err != db.ErrNoRows {
+			panic(err)
+		}
+		return file_status{file: File{path, info}, new: true}
+	}
+}
+
+func DryBackup(rawPaths []string) {
+	paths := make([]string, 0)
+	fileInfos := make([]os.FileInfo, 0)
```

Comment:
instead of making two different arrays of `[]string` and `[]os.FileInfo` let's just have one array of `[]File` that contains both